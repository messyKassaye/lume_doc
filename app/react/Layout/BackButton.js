import PropTypes from 'prop-types';
import React from 'react';

import { Icon } from 'UI';
import { I18NLink, t } from 'app/I18N';

const BackButton = ({ to }) => (
  <I18NLink to={to} className="btn btn-default">
    <Icon icon="arrow-left" directionAware />
    <span className="btn-label">{t('System', 'Back')}</span>
  </I18NLink>
);

BackButton.defaultProps = {
  to: '',
};

BackButton.propTypes = {
  to: PropTypes.string,
};

export default BackButton;
