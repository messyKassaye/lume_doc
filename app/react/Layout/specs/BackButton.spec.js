import React from 'react';
import { shallow } from 'enzyme';

import BackButton from '../BackButton';

describe('Icon', () => {
  let component;
  let props;

  const render = () => {
    component = shallow(<BackButton {...props} />);
  };

  it('should render the back button to the provided url', () => {
    props = { to: '/some/url' };
    render();
    expect(component).toMatchSnapshot();
  });
});
