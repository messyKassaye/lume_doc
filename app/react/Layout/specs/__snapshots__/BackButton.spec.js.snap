// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render the back button to the provided url 1`] = `
<Connect(I18NLink)
  className="btn btn-default"
  to="/some/url"
>
  <Connect(Icon)
    directionAware={true}
    icon="arrow-left"
  />
  <span
    className="btn-label"
  >
    Back
  </span>
</Connect(I18NLink)>
`;
